pages/stands/index.tsx

```diff
@@ -0,0 +1,299 @@
+
+// Update the currentEvent-list every 30 sec
+const intervalLength = 1000 * 30;
+
+type FeaturedEvent = ArrayElement<
+  NonNullable<
+    NonNullable<
+      ArrayElement<NonNullable<stands_QueryResponse['stands']>>
+    >['events']
+  >
+>;
+
+type Companies =
+  | stands_QueryResponse['currentMetaData']['companiesFirstDay']
+  | stands_QueryResponse['currentMetaData']['companiesLastDay']
+  | stands_QueryResponse['currentMetaData']['collaborators'];
+
+const getFeaturedEventStands = (
+  time: Dayjs,
+  stands: stands_QueryResponse['stands'],
+  startDate: string,
+  endDate: string,
+  companiesFirstDay: Companies,
+  companiesLastDay: Companies
+): stands_QueryResponse['stands'] => {
+  const featuredStands = stands?.filter(
+    (stand) =>
+      stand &&
+      currentDayCompaniesIds(
+        startDate,
+        endDate,
+        companiesFirstDay,
+        companiesLastDay
+      ).includes(stand.company.id) &&
+      currentFeaturedEvent(time, stand)
+  );
+  return featuredStands ?? [];
+};
+
+const currentDayCompaniesIds = (
+  startDate: string,
+  endDate: string,
+  companiesFirstDay: Companies,
+  companiesLastDay: Companies
+): string[] => {
+  if (isRespectiveDate(toDayjs(startDate))) {
+    return companyIds(companiesFirstDay);
+  } else if (isRespectiveDate(toDayjs(endDate))) {
+    return companyIds(companiesLastDay);
+  }
+  return [];
+};
+
+export const currentFeaturedEvent = (
+  time: Dayjs,
+  stand: ArrayElement<NonNullable<stands_QueryResponse['stands']>>
+): FeaturedEvent | null => {
+  const featuredEvent = stand?.events?.find(
+    (event) =>
+      event.type === 'A_7' &&
+      timeIsBetween({
+        time: time,
+        start: toDayjs(event.date, event.timeStart),
+        end: toDayjs(event.date, event.timeEnd),
+      })
+  );
+
+  return featuredEvent ?? null;
+};
+
+const companyIds = (companies: Companies): string[] =>
+  companies ? companies.map((company) => company.id) : [];
+
+const Index = ({
+  error,
+  props,
+}: WithDataAndLayoutProps<stands_QueryResponse>): JSX.Element => {
+  const [time, setTime] = useState(dayjs());
+  const [featuredEventStands, setFeaturedEventStands] = useState<
+    stands_QueryResponse['stands']
+  >([]);
+
+  const {
+    mainCollaborator,
+    collaborators,
+    companiesFirstDay,
+    companiesLastDay,
+    startDate,
+    endDate,
+  } = props.currentMetaData;
+
+  useEffect(() => {
+    const interval = setInterval(() => setTime(dayjs()), intervalLength);
+    return (): void => clearInterval(interval);
+  }, [props.currentMetaData]);
+
+  useEffect(() => {
+    const updatedFeaturedStands = getFeaturedEventStands(
+      time,
+      props.stands,
+      startDate,
+      endDate,
+      companiesFirstDay,
+      companiesLastDay
+    );
+    if (updatedFeaturedStands) {
+      setFeaturedEventStands(updatedFeaturedStands);
+    }
+  }, [
+    time,
+    props.stands,
+    startDate,
+    endDate,
+    companiesFirstDay,
+    companiesLastDay,
+  ]);
```

Comment:
Same as above here. Do you need these hooks? and the featuredEventStands state?

Reply:
I see the interval hook is needed, but the second one can be removed as far as I can tell

Reply:
The thought behind it is that in the change from 12:59-13:00 would rerender the new featuredStands. In my mind that wouldn't happen if not within a useEffect?

Reply:
The state change for `time` will cause a rerender and call getFeaturedStands again. I tested this out to be sure :smile: